x-pack/plugins/case/server/client/cases/utils.ts

```diff
@@ -8,127 +8,176 @@
 import { i18n } from '@kbn/i18n';
 import { flow } from 'lodash';
 import {
-  ServiceConnectorCaseParams,
-  ServiceConnectorCommentParams,
+  ActionConnector,
+  CaseResponse,
+  CaseFullExternalService,
+  CaseUserActionsResponse,
+  CommentResponse,
+  CommentResponseAlertsType,
+  CommentType,
   ConnectorMappingsAttributes,
   ConnectorTypes,
+  CommentAttributes,
+  CommentRequestUserType,
+  CommentRequestAlertType,
+} from '../../../common/api';
+import { ActionsClient } from '../../../../actions/server';
+import { externalServiceFormatters, FormatterConnectorTypes } from '../../connectors';
+import { CaseClientGetAlertsResponse } from '../../client/alerts/types';
+import { isUserContext } from '../../routes/api/utils';
+import {
+  BasicParams,
   EntityInformation,
   ExternalServiceParams,
+  ExternalServiceComment,
   Incident,
-  JiraPushToServiceApiParams,
   MapIncident,
   PipedField,
   PrepareFieldsForTransformArgs,
   PushToServiceApiParams,
-  ResilientPushToServiceApiParams,
-  ServiceNowITSMPushToServiceApiParams,
-  SimpleComment,
   Transformer,
   TransformerArgs,
   TransformFieldsArgs,
-} from '../../../../../common/api';
-import { ActionsClient } from '../../../../../../actions/server';
-export const mapIncident = async (
-  actionsClient: ActionsClient,
+} from './types';
+
+export const getLatestPushInfo = (
   connectorId: string,
-  connectorType: string,
-  mappings: ConnectorMappingsAttributes[],
-  params: ServiceConnectorCaseParams
-): Promise<MapIncident> => {
-  const { comments: caseComments, externalId } = params;
+  userActions: CaseUserActionsResponse
+): { index: number; pushedInfo: CaseFullExternalService } | null => {
+  for (const [index, action] of [...userActions].reverse().entries()) {
```

Comment:
Since we moved the logic to the back end, I really do think that we can do this differently by querying of savedObject with filter
```
{
   "size": 1,
   "sort": { "action_at": "desc"},
 ...
 }
```

Reply:
That's a nice idea! Is it possible to have the position of the action in the list of userActions? I want to filter all comments added after the push. Maybe we can do two queries, one that retrieves the `push` user action and then filter the comments based on the `action.created_at` attribute. What do you think?

Reply:
Let's do that in another PR to no block @jonathan-buttner, I do think that you should be able to minimize the amount of data by filtering by `"action": "push-to-service"` and `"sort": { "action_at": "desc"}`. I will feel better by doing that and using less memory on the kibana node side. Please do a follow up PR, I think that's really iomportant to us and our user

Reply:
Ok! You are right!